mwdb/model/object.py

```diff
@@ -892,6 +901,48 @@ def remove_attribute(self, key, value, check_permissions=True):
                 raise
         return True
 
+    def get_uploaders(self):
+        """
+        Gets all object uploads visible for currently authenticated user
+        :rtype: List[ObjectPermission]
+        """
+
+        from .group import Group, Member
+
+        # Users with access_uploader_info can see every upload
+        if g.auth_user.has_rights(Capabilities.access_uploader_info):
+            uploaders = (
+                db.session.query(ObjectPermission)
+                .filter(ObjectPermission.object_id == self.id)
+                .filter(ObjectPermission.reason_type == "added")
+                .order_by(ObjectPermission.access_time.asc())
```

Comment:
1. Use AccessType constants to make it easier to find references
2. I think that `get_uploaders` should return only direct uploaders excluding inherited ADDED entries

```suggestion
                .filter(ObjectPermission.object_id == self.id)
                .filter(ObjectPermission.reason_type == AccessType.ADDED)
                .filter(ObjectPermission.related_object_id == self.id)
                .order_by(ObjectPermission.access_time.asc())
```